Fixed Spektrum binding when overclocked.

mikeller opened the pull request.

brucesdad13 commented:
You're the best!

AndersHoglund commented:
Duplicate of #4132, that was reverted by #4141. What is happening??
Please remove the `spektrum_sat_bind_autoreset` change and keep the OC correction, then this PR will be fine. Thought we agreed on this in the two prev PRs.

martinbudden commented:
@mikeller , @AndersHoglund , what is the status of this PR?

mikeller commented:
In my opinion it's good to go, but @AndersHoglund seems to prefer to have it in 3.3.

AndersHoglund commented:
Remove the second "Also changed ...." -part. The first OC fix part of this PR is good to go.

mikeller commented:
Fixed.

AndersHoglund commented:
C'mon. Not only the PR title text. The change in `settings.c` is still there. Now we have a change without description, even worse.

mikeller commented:
No. As I already told you, I am not going to go out of my way to please people who seem to be only around to pull other people's work down. If you want this in 3.3 (which seemed to be what you were complaining about in the first pull request for this), I am fine with this.

AndersHoglund commented:
You suggested 3.3, and OK I did agree on that since that part is a UI change, affecting end users. The OC fix could/should be done now in 3.2 Two totally different things mixed up in one PR. It is a principle not to do that. As stated in the original PR, no big deal, as this is only a one line change. But the principle is applicable, one line or 1000 lines. Consider bisect trouble shooting, you need to see one issue change per commit. Or as it happened when you reverted the original PR, then both issues was reverted. Even though only one was reversal requested. 

If you still want those two changes done in one PR, both should be described. It is either or, change and description. Or no change (and no description). Whatever, I give up.

mikeller commented:
@AndersHoglund: While I agree that pull requests should not cover totally unrelated topics or actions, I think your demand that multiple bugs affecting the same functionality (Spektrum bind in this case) goes over the top. I agree that API changes like the fix of the data type of `spektrum_sat_bind_autoreset` should be documented, and so I reverted the first PR and resubmitted with an appropriate title, in order to get it picked up in the generated release notes.

Changed back the title to reflect the API, this is ready to go now, @martinbudden.

AndersHoglund commented:
Thank you for understanding and agree-ing. On a higher level all issues and PRs are related in some way, even if totally separate on the low level.
